### DISCUSSION TITLE
How to have initial mints to owner?

### DISCUSSION THREAD

**Discussion by JohnCarp** [hashlips-lab/nft-erc721-collection#450] [Q&A]
Hey @liarco! 

So I'm not sure this is a bug or not but one of the comments mentioned this was not working? 

https://github.com/hashlips-lab/nft-erc721-collection/issues/39#issuecomment-1153163518

I seen you left this comment previously to that:

https://github.com/hashlips-lab/nft-erc721-collection/issues/39#issuecomment-1064064456

Essentially I would like to mint a certain amount like 100 to keep for the team. A couple questions on this:

* Is this code working or is there really a bug?
* Does this happen when the contract is deployed?
* If not does it happen when we make the mint live?

Thank you for the help and apologies if this was not a bug!

**Comment by mewmix** [comment]
This constructor minted 100 nfts on my testing , per the author's referenced comments. Would advise to test on your own for further safety.


```solidity
  constructor(
    string memory _tokenName,
    string memory _tokenSymbol,
    uint256 _cost,
    uint256 _maxSupply,
    uint256 _maxMintAmountPerTx,
    string memory _hiddenMetadataUri
  ) ERC721A(_tokenName, _tokenSymbol) {
    setCost(_cost);
    maxSupply = _maxSupply;
    setMaxMintAmountPerTx(_maxMintAmountPerTx);
    setHiddenMetadataUri(_hiddenMetadataUri);
    _safeMint(msg.sender, 100);

  }
```

**Comment by liarco** [comment]
Thank you very much @mewmix for you help. I have to move this to discussions since it's not really a bug report.

**Comment by batmanbury** [comment]
@mewmix Just out of curiosity, I tried this twice, but it's not minting anything to the owner's wallet for me. How did it work? Did you have to confirm multiple transactions maybe? One for deployment, then again for the mints?

**Reply by batmanbury** [reply]
Update: Actually I see on etherscan that the totalSupply in the Read Contract is the amount I had set in _safeMint in the constructor, so it seems it did work, but they weren't minted to my wallet. Should they be? Or are they in the smart contract itself?

**Reply by batmanbury** [reply]
Nevermind! I believe it works correctly, it's just that metamask wallet doesn't display them, and then Opensea doesn't support Sepolia testnet. But they are there.

**Reply by liarco** [reply]
Hi @batmanbury, you are correct! A quick tip: if you want to test OpenSea properly you should use Goerli (I know gas prices are insane on that testnet atm), otherwise you can use Mumbai which is the testnet for Polygon so not quite the same as Goerli/Sepolia, but the EVM runtime behaves exactly the same with our contract (unless you made big changes to the source).

I hope this helps and I wish you all the best for your project 😉
